File: BasicClassValidation/Classes/Movie.vb
Imports System.ComponentModel.DataAnnotations

Namespace Classes

    Public Class Movie
        Public Property Id As Integer

        <StringLength(60, MinimumLength:=3)>
        Public Property Title As String

        <DataType(DataType.Date)>
        Public Property ReleaseDate As DateTime

        <RegularExpression("^[A-Z]+[a-zA-Z''-'\s]*$")>
        <Required>
        <StringLength(30)>
        Public Property Genre As String

        <DataType(DataType.Currency)>
        <DisplayFormat(DataFormatString:="{0:c}")>
        <Required(ErrorMessage:="Price is Required"),
            Range(10, 100,
                  ErrorMessage:="{0} must be between {1} and {2} dollars.")>
        Public Property Price As Decimal

        <RegularExpression("^[A-Z]+[a-zA-Z''-'\s]*$")>
        <StringLength(4)>
        Public Property Rating As String
    End Class
End Namespace

File: ValidationSample/Models/Movie.cs
using System;
using System.ComponentModel.DataAnnotations;
using ValidationSample.Validation;

namespace ValidationSample.Models
{

    public class Movie
    {
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Title { get; set; }

        [ClassicMovie(1960)]
        [DataType(DataType.Date)]
        [Display(Name = "Release Date")]
        public DateTime ReleaseDate { get; set; }

        [Required(ErrorMessage = "The {0} is required")]
        [StringLength(1000)]
        public string Description { get; set; }

        [Range(10, 999.99)]
        public decimal Price { get; set; }

        public Genre Genre { get; set; }

        public bool PreOrder { get; set; }
    }

}
